samtranslator/model/api/api_generator.py

```diff
@@ -391,9 +391,11 @@ def _add_cors(self):
                                                             "'AllowOrigin' is \"'*'\" or not set")
 
         editor = SwaggerEditor(self.definition_body)
+        openapi_flag = True if self.definition_body.get("openapi") else False
```

Comment:
I think it's safer to remove produces and consumes in `_openapi_postprocess` than doing it in swagger. So instead of passing the flag to swagger, just add this to the method `_openapi_postprocess` here so we don't have too many openapi/swagger if checks in swagger.py, and we also make sure we chack not only for the presence of the flag but slso that the version is 3.x.x